### PR TITLE
Pass originalEvent when custom event maps to interaction

### LINKED ISSUE #126: Pass originalEvent for all custom events with direct mapping to DOM event

### PR DESCRIPTION
**Pull request by yomed**
## Description
- adds `originalEvent` for breadcrumb, carousel, and pagination
- update docs
- includes minor cleanup and related refactoring 

## Context
I looked through each component for events that only map to user interaction, and then added `originalEvent` to the custom event data. There are some cases where the event can be triggered by both UI and API. For those cases, I am not passing the `originalEvent`, because I think it seems strange to sometimes pass it and sometimes not for the same event. We can see if teams actually need this functionality later on.

## References
Follow up to #141 
Fixes #126

### CODE REVIEW

File: src/components/ebay-breadcrumb/index.js [hunk 1]
```diff
@@ -45,9 +45,9 @@ function getInitialState({ hijax }) {
     return { hijax };
 }
 
-function handleClick(event) {
-    eventUtils.preventDefaultIfHijax(event, this.state.hijax);
-    emitAndFire(this, 'breadcrumb-select', { el: event.target });
+function handleClick(originalEvent) {
+    eventUtils.preventDefaultIfHijax(originalEvent, this.state.hijax);
+    emitAndFire(this, 'breadcrumb-select', { originalEvent, el: event.target });
```
Comment (PrashantAshok) [inline src/components/ebay-breadcrumb/index.js]: i think you meant `originalEvent.target`
```emitAndFire(this, 'breadcrumb-select', { originalEvent, el: originalEvent.target });```

Comment (seangates) [reply src/components/ebay-breadcrumb/index.js]: Yup. I think it should be `originalEvent.target`, too.

**Comment by ianmcburnie** [comment]
Looks like there are some conflicts that need resolving.

**Comment by ianmcburnie** [comment]
`There are some cases where the event can be triggered by both UI and API`

This gets me back to thinking that we may need to more explicitly separate and name some of our events. For example, `pagination-next-click` which is clearly UI based, as opposed to `pagination-next` which is API based.

**Comment by yomed** [comment]
Maybe, though I don't think there's any rush on this. A more apt example now might be `notice-hide`, which can be triggered from both UI and API. It's possible that users would need `originalEvent`, but depending on what the reasoning is, we might not send that along anyway. IMO it's hard to gauge this right now since we don't have a lot of cases for either API usage, or `originalEvent` usage.

**Comment by senthilp** [comment]
Agree, for now, we should be ok. We just need to follow this guideline — if there is a native event (UI triggered), `originalEvent` has to be passed.

**Comment by ianmcburnie** [comment]
Yep, no rush. I think we'll have it all figured out by v1.0.0 (i.e. the grand opening).

**Pull request merged**